azkaban-common/src/main/java/azkaban/executor/ExecutionController.java

```diff
@@ -488,13 +488,7 @@ public LogData getExecutionJobLog(final ExecutableFlow exFlow, final String jobI
   @Override
   public Map<String, String> getExternalJobLogUrls(final ExecutableFlow exFlow, final String jobId,
       final int attempt) {
-
     final Map<String, String> jobLogUrlsByAppId = new LinkedHashMap<>();
-    if (!this.azkProps.containsKey(ConfigurationKeys.RESOURCE_MANAGER_JOB_URL) ||
-        !this.azkProps.containsKey(ConfigurationKeys.HISTORY_SERVER_JOB_URL) ||
```

Comment:
How will it work incase of no cluster definition or UNKOWN?

Reply:
If no cluster definition is found, it fall backs to loading the link URLs from Azkaban Web server configuration. This preserves the current behavior. 

For an UNKNOWN cluster, because it does not have URLs defined, so ExecutionControllerUtils. createJobLinkUrl() return null.